File: Binary_Tree_Postorder_Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> postorderTraversal(TreeNode *root) {
        // IMPORTANT: Please reset any member data you declared, as
        // the same Solution instance will be reused for each test case.
        vector<int> ret;
        if (root == NULL) return ret;
        stack<TreeNode*>st;
        st.push(root);
        while (!st.empty()){
            TreeNode *node = st.top();
            st.pop();
            ret.insert(ret.begin(), node->val);
            if (node->left != NULL){
                st.push(node->left);
            }
            if (node->right != NULL) {
                st.push(node->right);
            }
        }
        return ret;
    }
};

File: Java/Binary_Tree_Postorder_Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ArrayList<Integer> postorderTraversal(TreeNode root) {
        ArrayList<Integer> ret = new ArrayList<Integer>();
        Stack<TreeNode> st = new Stack<TreeNode>();
        if (root == null) return ret;
        st.push(root);
        TreeNode node = null;
        while (!st.isEmpty()) {
            node = st.pop();
            ret.add(0, node.val);
            if (node.left != null) st.push(node.left);
            if (node.right != null) st.push(node.right);
        }
        return ret;
    }
}
